api.planx.uk/editor/publish.ts

```diff
@@ -48,6 +49,18 @@ const publishFlow = async (
 
   try {
     const flattenedFlow = await dataMerged(req.params.flowId);
+
+    // Check the flattened flow to ensure that all sections (type 360) are still in valid positions (eg within `_root: { edges: [...]}`, not the edge of a portal)
+    //   If any are not, don't proceed with publishing and return a message to display in the Editor
+    const sectionTypeNodeIds = Object.entries(flattenedFlow).filter(([_nodeId, nodeData]) => nodeData?.type === 360).map(([nodeId, _nodeData]) => nodeId);
+    const intersectingNodeIds = intersection(flattenedFlow["_root"].edges, sectionTypeNodeIds);
+    if (intersectingNodeIds.length !== sectionTypeNodeIds.length) {
+      return res.json({
+        alteredNodes: null,
+        message: "Error publishing: found Sections in one or more External Portals"
+      });
```

Comment:
I opted to not actually return a true HTTP error here, but rather return a message that is shown to editors. Once you adjust your content, clicking "publish" will succeed again
![Screenshot from 2023-02-23 11-37-39](https://user-images.githubusercontent.com/5132349/220884459-30479a70-bb26-4f4b-8ce7-e1e2aad190da.png)

I also imagine we'll definitely want to rethink how/where "validation checks" happen during publishing once we have > 1 ! But checking directly in the publish method feels okay to me for this first, rather simple, check.

But open to suggestions here!

Reply:
On the HTTP error part, returning success with null `alteredNodes` & a `message` is currently consistent with what happens when the publish endpoint doesn't detect any changes to publish (which is also a sort of fuzzy success or error scenario in that the actual publish mutation is skipped!)

If/when we start throwing actual errors, we'll have to rework how `setLastPublishedTitle` works in the preview browser which controls messages shown to editors, which feels a bit bigger scope than here but a worthwhile future rethink & todo for sure! More related to publishing/editor work in general than sections in my opinion